src/Microsoft.DotNet.Core.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -0,0 +1,270 @@
+
+            var libraryLookup = lockFile.Libraries.ToDictionary(l => l.Name, StringComparer.OrdinalIgnoreCase);
+
+            var runtimeSignature = GenerateRuntimeSignature(runtimeExports);
+
+            IEnumerable<RuntimeLibrary> runtimeLibraries =
+                new[] { GetProjectLibrary(projectName, projectVersion, lockFile, lockFileTarget, dependencyLookup) }
+                .Concat(GetLibraries(runtimeExports, libraryLookup, dependencyLookup, runtime: true).Cast<RuntimeLibrary>());
+
+            return new DependencyContext(
+                new TargetInfo(framework.DotNetFrameworkName, runtime, runtimeSignature, portable),
+                compilerOptions ?? CompilationOptions.Default,
+                Enumerable.Empty<CompilationLibrary>(), //GetLibraries(compilationExports, dependencyLookup, runtime: false).Cast<CompilationLibrary>(),
+                runtimeLibraries,
+                new RuntimeFallbacks[] { });
+        }
+
+        private static IEnumerable<LockFileTargetLibrary> FilterPlatformDependencies(
+            IEnumerable<LockFileTargetLibrary> runtimeExports,
+            LockFileTargetLibrary platformExport)
+        {
+            var exportsLookup = runtimeExports.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+
+            HashSet<string> exclusionList = GetPlatformExclusionList(platformExport, exportsLookup);
+
+            return runtimeExports.Where(e => !exclusionList.Contains(e.Name));
+        }
+
+        private static HashSet<string> GetPlatformExclusionList(
+            LockFileTargetLibrary platformExport,
+            IDictionary<string, LockFileTargetLibrary> exportsLookup)
+        {
+            var exclusionList = new HashSet<string>();
+
+            exclusionList.Add(platformExport.Name);
+            CollectDependencies(exportsLookup, platformExport.Dependencies, exclusionList);
+
+            return exclusionList;
+        }
+
+        private static void CollectDependencies(
+            IDictionary<string, LockFileTargetLibrary> exportsLookup,
+            IEnumerable<PackageDependency> dependencies,
+            HashSet<string> exclusionList)
+        {
+            foreach (PackageDependency dependency in dependencies)
+            {
+                LockFileTargetLibrary export = exportsLookup[dependency.Id];
+                if (export.Version.Equals(dependency.VersionRange.MinVersion))
+                {
+                    exclusionList.Add(export.Name);
+                    CollectDependencies(exportsLookup, export.Dependencies, exclusionList);
+                }
+            }
+        }
+
+        private static string GenerateRuntimeSignature(IEnumerable<LockFileTargetLibrary> runtimeExports)
+        {
+            var sha1 = SHA1.Create();
+            var builder = new StringBuilder();
+            var packages = runtimeExports
+                .Where(libraryExport => libraryExport.Type == "package");
+            var seperator = "|";
```

Comment:
I think this makes the code less readable, so when there is only a single hard-coded string I leave it inline.